docs/guides/deployment-patterns/kubernetes/kubernetes-deployment.md

```diff
@@ -0,0 +1,223 @@
+# Kubernetes Deployment Guide
+
+This guide provides comprehensive instructions for deploying Thunder in Kubernetes environments using Helm charts, covering everything from prerequisites to production configurations.
+
+## Architecture Overview
+
+![Thunder Kubernetes Architecture](thunder-kubernetes-diagram.png)
+
+The diagram above shows the Thunder deployment architecture in Kubernetes, including the main application pods, ingress controller, and database configuration options.
+
+## Table of Contents
+
+- [Prerequisites](#prerequisites)
+- [Quick Start](#quick-start)
+- [Installation Methods](#installation-methods)
+- [Database Setup](#database-setup)
+
+## Prerequisites
+
+### Infrastructure Requirements
+
+- **Kubernetes Cluster**: A running Kubernetes cluster (v1.19+)
+  - [minikube](https://kubernetes.io/docs/tasks/tools/#minikube) for local development
+  - [kind](https://kind.sigs.k8s.io/) for local testing
+  - Managed Kubernetes services (EKS, GKE, AKS) for production
+- **Ingress Controller**: NGINX Ingress Controller (recommended) or alternative
+- **TLS Certificates**: Valid SSL certificates for production deployments
+
+### Required Tools
+
+| Tool          | Installation Guide | Version Check Command|
+|---------------|--------------------|-----------------------|
+| Git           | [Install Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) | `git --version` |
+| Helm          | [Install Helm](https://helm.sh/docs/intro/install/) | `helm version` |
+| kubectl       | [Install kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) | `kubectl version` |
+| Docker        | [Install Docker](https://docs.docker.com/engine/install/) | `docker --version` |
+
+### Verify Prerequisites
+
+```bash
+# Check Kubernetes cluster access
+kubectl cluster-info
+
+# Verify Helm installation
+helm version
+
+# Check ingress controller (If available)
+kubectl get pods -n ingress-nginx
+```
+
+## Quick Start
+
+Deploy Thunder with default settings in under 5 minutes:
+
+
+
+### 1. Install Thunder
+
+```bash
+# Install Thunder from GitHub Container Registry
+helm install thunder oci://ghcr.io/asgardeo/helm-charts/thunder
+
+# Or install a specific version
+helm install thunder oci://ghcr.io/asgardeo/helm-charts/thunder --version 0.11.0
```

Comment:
Is there a way to specify the latest without mentioning a specific version?

Reply:
helm install thunder oci://ghcr.io/asgardeo/helm-charts/thunder

We can use this. line:61